src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -316,11 +331,95 @@ g.test('texture_size,2d_texture')
     }, !success);
   });
 
-g.test('texture_size,3d_texture')
-  .desc(
-    `Test texture size requirement for 3D texture.
-	TODO: add tests for compressed texture.`
+g.test('texture_size,2d_texture,compressed_format')
+  .desc(`Test texture size requirement for 2D texture with compressed format.`)
+  .subcases(() =>
+    params()
+      .combine(poptions('format', kCompressedTextureFormats))
+      .combine(poptions('dimension', [undefined, '2d'] as const))
+      .combine([
```

Comment:
[1] this might need to become:

```
 .expand(p => {
     const {blockWidth, blockHeight} = kTextureFormatInfo[p.format];
     return poptions('size', [ same list that you have now ])
})
```

Reply:
Thanks. Looks like we can access parameters only in `unless`, `filter`, `expand` and something like that. But we can't access  previous parameters in another `.combine`.